### ISSUE TITLE
Navigation: view rotation in trackpad mode continues after rotation key released (unknown upstream Qt issue is source of problem)

### ISSUE THREAD

**Issue by inieves** [FreeCAD/FreeCAD#16824]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

In FC 0.21.2 on MacOS 14.7
In Touchpad mode

Expected Behavior:
After "option" key is released after rotating view, moving cursor does not rotate the view.

Actual Behavior:
After "option" key is released after rotating view, moving cursor does rotate the view slightly.

### Full version info

```shell
OS: macOS 14.7
Word size of FreeCAD: 64-bit
Version: 0.21.2.33771 (Git)
Build type: Release
Branch: (HEAD detached at 0.21.2)
Hash: b9bfa5c5507506e4515816414cd27f4851d00489
Python 3.10.13, Qt 5.15.8, Coin 4.0.0, Vtk 9.2.6, OCC 7.6.3
Locale: C/Default (C)
Installed mods: 
  * A2plus 0.4.64a
  * fasteners 0.5.20
```

### Subproject(s) affected?

None

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by luzpaz** [comment]
Please test on RC2 (https://github.com/FreeCAD/FreeCAD-Bundle/releases/tag/1.0rc2)

**Comment by hyarion** [comment]
I can confirm it on RC2 on mac.
<details><summary>Version info: 1.0.0RC2.38806 - macos14.5 </summary>
<p>
OS: macOS 14.5
Word size of FreeCAD: 64-bit
Version: 1.0.0RC2.38806 (Git)
Build type: Release
Branch: (HEAD detached at 1.0rc2)
Hash: 3d63fc6c2f665a8d5e6468845a419bcac80756c7
Python 3.11.9, Qt 5.15.13, Coin 4.0.3, Vtk 9.2.6, OCC 7.7.2
Locale: C/Default (C)
Stylesheet/Theme/QtStyle: FreeCAD Dark.qss/FreeCAD Dark/Fusion
Installed mods: 
  * CurvedShapes 1.0.10
  * sheetmetal 0.4.26
  * Curves 0.6.6

</p>
</details> 

https://github.com/user-attachments/assets/08fb1ea9-2b4d-419d-b45c-f76c514f5a6a

In this screen recording above, you will see the following:
1. the mouse is entering the view in the bottom right corner
2. option (alt on a non-mac) is pressed and not released
3. the mouse is moved and the view starts to rotate, the cursor also changes to a rotate cursor
4. the mouse stops moving and the rotation stops as well
5. option (alt) is now released, but the red dot indicating the rotation center is still on screen
6. the mouse is quickly moved to the left. The view is rotated one final step.
7. video is over.

What I think is happening is that the rotation mode is not deactivated when the modifier key is released. Instead it deactivates after the mouse event. This would explain why the red dot is showing when key is released.

The red dot doesn't show before you start moving the cursor either, which looks like the same issue is there too.

---

While this clearly is a bug, the impact is minimal. I hadn't noticed it before.

A workaround is to give FreeCAD another mouse event, like a pressing a mouse button (or clicking on the track pad). That will abort the rotation state so the view doesn't jump when mouse is moved afterwards. 

@inieves Is this the behaviour you see as well?

**Comment by Rexbas** [comment]
I can reproduce this. I am already looking at the Touchpad nav style for another small problem. I will try to look for a fix for this too.

**Comment by inieves** [comment]
Yes this is the behavior I see.

**Comment by Rexbas** [comment]
It looks like `View3DInventorViewer::processSoEvent(const SoEvent* ev)` is never called when the event is a `SoKeyboardEvent` triggered by the left alt key. So when it is pressed, it won't immediately go into dragging mode and when it is released it won't go to idle mode. Only after another key or mouse event was triggered it syncs the state of the alt button but this may already be too late which is why we see the view move.

I think the menu bar already processes the alt key and then it does not propagate anymore?! Not sure how to fix this.

**Comment by hyarion** [comment]
I tried setting a breakpoint on `src/Gui/Quarter/Keyboard.cpp` line 75: `return PRIVATE(this)->keyEvent((QKeyEvent *) event);`
The program stops on break point for Shift, Command, and Ctrl (this is on macOS), but not on option(alt).

This method is responsible to convert all qt keyboard events to coin counterparts. As the event hasn't reached coin yet, it can't be coin related. It must be Qt related as @Rexbas mentions above.

**Comment by FEA-eng** [comment]
@inieves or @maxwxyz Can you check if it's still relevant on Mac?

**Comment by inieves** [comment]
I cannot check this in immediate future, i am away from workstation.  I will check as soon as possible though.-iOn Oct 21, 2025, at 10:36 AM, FEA-eng ***@***.***> wrote:﻿FEA-eng left a comment (FreeCAD/FreeCAD#16824)
@inieves or @maxwxyz Can you check if it's still relevant on Mac?

**Comment by inieves** [comment]
I can confirm the bug is still relevant on Mac.
I tested FC 1.0.2.

Best,
Ian

**Comment by luzpaz** [comment]
@inieves can you test on development build ? https://github.com/FreeCAD/FreeCAD/releases

**Comment by inieves** [comment]
i verified 2025.11.05 does not pass

i cannot test 2025.11.12, macos download not available

**Comment by luzpaz** [comment]
> i cannot test 2025.11.12, macos download not available

@inieves it is now. https://github.com/FreeCAD/FreeCAD/releases click the 'Assets' to expand the subsection.

**Comment by inieves** [comment]
i verified 2025.11.12 does not pass

**Comment by luzpaz** [comment]
@inieves can you test on 1.1RC2 and 1.2.0-dev ? https://github.com/FreeCAD/FreeCAD/releases

@johnocfii can you reproduce on macOS ?

**Comment by JohnOCFII** [comment]
> [@JohnOCFII](https://github.com/JohnOCFII) can you reproduce on macOS ?

Yes.  I can reproduce the issue on both 1.1rc2 and a fresh build of 1.2.0-dev.

```
OS: macOS 26.2
Architecture: arm64
Version: 1.2.0dev.45082 (Git)
Build date: 2026/01/20 20:22:29
Build type: Unknown
Branch: main
Hash: dee4ab6dbd844e76309ddde5a3def638d28dc789
Python 3.11.13, Qt 6.8.3, Coin 4.0.3, Vtk 9.3.1, boost 1_86, Eigen3 3.4.0, PySide 6.8.3
shiboken 6.8.3, xerces-c 3.3.0, IfcOpenShell 0.8.0, OCC 7.8.1
Locale: English/United States (en_US)
Navigation Style/Orbit Style/Rotation Mode: Touchpad/Rounded Arcball/Window center
Stylesheet/Theme/QtStyle: OpenDark.qss/OpenDark/
Logical DPI/Physical DPI/Pixel Ratio: 72/127.5/2
Installed mods: 
  * OpenTheme 2025.5.20
  * Render 2024.12.15
  * Telemetry 1.0.4
  * Curves 0.6.70
```
